### PR TITLE
Add WeiKai component documentation

### PR DESCRIPTION
**Pull request by DrCoolzic**
Documentation for the new Weikai family of components

## Description:
This component adds support for the Weikai family of component chips. It add the following components:
wk2132_spi, wk2212_spi, wk2204_spi, wk2168_spi, wk2132_i2c, wk2212_i2c, wk2204_i2c, wk2168_i2c

The code for the component has been submitted with the PR #5218

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5218
Add WK2132 component to esphome #5218
## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by DrCoolzic** [comment]
PR for code
Add WK2132 component to esphome #5218

File: components/wk2132_i2c.rst [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+The WK2132 I²C component (wk2132_i2c) allows you to use a 
+`Weika WK2332-ISSG <https://jlcpcb.com/partdetail/Weikai-WK2132ISSG/C401039>`__
+chip or a DFROBOT `Gravity: I²C to Dual UART Module <https://www.dfrobot.com/product-2001.html>`__
+module connected to an ESP32 MCU through an I²C bus. 
+
+The WK2132 chip/module serves as an I²C to Dual UART converter, supporting data transmission rates 
+of up to 1 Mbps. While the chip can also interface with an SPI bus, this functionality is supported by 
+a separate ESPHome component (wk2132_spi). Each UART channel features two independent 256-byte FIFO hardware 
+buffers for transmission and reception.
```
Comment (jesserockz) [inline components/wk2132_i2c.rst]: Until such a time as that component exists, it is best not to document it as existing...
```suggestion
The WK2132 chip/module serves as an I²C to Dual UART converter, supporting data transmission rates 
of up to 1 Mbps. Each UART channel features two independent 256-byte FIFO hardware 
buffers for transmission and reception.
```

File: components/wk2132_i2c.rst [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      platformio_options:
+        build_flags:
+          - -DI2C_BUFFER_LENGTH=256
+
+.. caution:: 
+
+    Currently, due to some internal limitations, it is necessary to declare a dummy ``uart:`` component in
+    your configuration **even if you do not need one**. See the example above. This requirement 
+    will disappear in future release.\ :raw-html-m2r:`<br>`
+
```
Comment (jesserockz) [inline components/wk2132_i2c.rst]: This limitation will be fixed before the component is merged.
```suggestion
```

File: components/wk2132_i2c.rst [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  - **parity** (*Optional*): The parity used on the UART channel. Options: ``NONE``, ``EVEN``, 
+    ``ODD``. Defaults to ``NONE``.
+  - **stop_bits** (*Optional*): The number of stop bits to send. Options: ``1``, ``2``. 
+    Defaults to ``1``.
+
+See Also
+********
+
+- :ref:`i2c`
+- :ghedit:`Edit`
```
Comment (jesserockz) [inline components/wk2132_i2c.rst]:
```suggestion
- :apiref:`wk2132_i2c/wk2132_i2c.h`
- :ghedit:`Edit`
```

File: components/wk2132_i2c.rst [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    Currently, due to some internal limitations, it is necessary to declare a dummy ``uart:`` component in
+    your configuration **even if you do not need one**. See the example above. This requirement 
+    will disappear in future release.\ :raw-html-m2r:`<br>`
+
+    Currently, there are bugs in the ESPHome I²C component and in the Arduino Wire library that limit the 
+    maximum transmission size to 255. The current version of wk2132_i2c automatically sets this limit for 
+    you. This limitation will disappear in future versions
+
+Component configuration variables:
+**********************************
```
Comment (jesserockz) [inline components/wk2132_i2c.rst]:
```suggestion
Configuration variables:
************************
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/wk2132_i2c.rst [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+    # Example of configuration entry
+    
+    i2c:
+      id: i2c_bus
+      scan: true
+      sda: 21
+      scl: 22
+      frequency: 600kHz
+
```
Comment (kbx81) [inline components/wk2132_i2c.rst]: Typically we don't include the configuration of dependent components in the example.
```suggestion
```

**Comment by DrCoolzic** [comment]
Sorry as the PR was still open I took the oportunity to squeeze the SPI version as well :)

**Comment by nagyrobi** [comment]
@DrCoolzic can you please check if @jesserockz 's observations are still valid or not, and resolve them if necessarry? (if already resolved, mark them as resolved please)

**Comment by DrCoolzic** [comment]
> @DrCoolzic can you please check if @jesserockz 's observations are still valid or not, and resolve them if necessarry? (if already resolved, mark them as resolved please)

I beleive I have now marked the request as resolved. I had to made a change to my rst file based on feedback from @jesserockz and I just pushed it few minutes ago. Let me know if that fulfill your request

**Comment by DrCoolzic** [comment]
Modified the documentation to reflect changes required by @jesserockz to uart_id
Plus miscellaneous spelling corrections

Ready for review :)

**Review by jesserockz — APPROVED** [review]

**Pull request merged**